docs/gallery/howto/autogen/error_resistant.py

```diff
@@ -0,0 +1,204 @@
+
+######################################################################
+# Load the AiiDA environment
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+
+from aiida_workgraph import WorkGraph, Task
+
+from aiida import load_profile, orm
+from aiida.common.exceptions import NotExistent
+from aiida.cmdline.utils.ascii_vis import format_call_graph
+
+load_profile()
+
+
+try:
+    bash_code = orm.load_code(
+        "bash@localhost"
+    )  # The computer label can also be omitted here
+except NotExistent:
+    bash_code = orm.InstalledCode(
+        label="bash",
+        computer=orm.load_computer("localhost"),
+        filepath_executable="/bin/bash",
+        default_calc_job_plugin="core.arithmetic.add",
+    ).store()
+
+
+######################################################################
+# Exit codes
+# ----------
+#
+# In AiiDA we can create error handlers for exit codes that are defined in the ``CalcJob`` specification.
+# If you are not familiar with the concept of exit codes in ``CalcJobs`` s you might want to familiarize yourself in the `aiida-core documentation <https://aiida.readthedocs.io/projects/aiida-core/en/latest/topics/processes/concepts.html#topics-processes-concepts-exit-codes>`__
+# For example for the ``ArithmeticAddCalculation`` we can the defined exit codes with
+#
+
+from aiida.calculations.arithmetic.add import ArithmeticAddCalculation
+
+print(ArithmeticAddCalculation.exit_codes)
+
+
+######################################################################
+# We will write an error handler for the exit code 410 ``ERROR_NEGATIVE_NUMBER``.
+#
+
+print(ArithmeticAddCalculation.exit_codes.ERROR_NEGATIVE_NUMBER)
+
+
+######################################################################
+# If the computed sum of the inputs x and y is negative, the ``ArithmeticAddCalculation`` fails with exit code 410.
+# We will run a calculation that exits with this error.
+#
+
+wg = WorkGraph("error_negative_number")
+wg.add_task(ArithmeticAddCalculation, name="add", x=1, y=-6, code=bash_code)
+
+
+wg.run()
+print("Task finished OK?:", wg.tasks.add.process.is_finished_ok)
+print("Exit code        :", wg.tasks.add.process.exit_code)
+print("Exit Message:    :", wg.tasks.add.process.exit_message)
+
+
+######################################################################
+# We can confirm that the task fails with this exit code in the CLI.
+#
+
+print(format_call_graph(orm.load_node(wg.pk)))
+
+
+######################################################################
+# Error handling
+# --------------
+#
+# To “register” a error handler for a WorkGraph, you simply define a function that takes the ``task`` as its arguments, and attach it as the ``error_handler`` of the ``WorkGraph``.
+# You can specify the tasks and their exit codes that should trigger the error handler, as well as the maximum number of retries for a task:
+#
+# .. code:: python
+#
+#    tasks={"add": {"exit_codes": [410],
+#                   "max_retries": 5}
+#          }
+#
```

Comment:
This breaks the reading consistency, and also the same dictionary appears again in the wg.add_error_handler(...) call. Consider removing it.